src/pages/index.js

```diff
@@ -0,0 +1,278 @@
+function setUpCloseButtons() {
+  document.querySelectorAll(".modal__close-btn").forEach((button) => {
+    const modal = button.closest(".modal");
+    button.addEventListener("click", () => closeModal(modal));
+  });
+}
+setUpCloseButtons();
+
+// Load user info and cards
+api
+  .getAppInfo()
+  .then(([cards, userInfo]) => {
+    profileAvatar.src = userInfo.avatar;
+    profileName.textContent = userInfo.name;
+    profileDescription.textContent = userInfo.about;
+
+    cards.reverse().forEach((cardData) => {
+      cardsList.prepend(createCardElement(cardData));
+    });
+  })
+  .catch((error) => console.error("Failed to load initial app info:", error));
+
+// Card creation function
+function createCardElement(data) {
+  const template = document.querySelector("#card-template");
+  const cardElement = template.content.querySelector(".card").cloneNode(true);
+
+  const cardImageEl = cardElement.querySelector(".card__image");
+  cardImageEl.src = data.link;
+  cardImageEl.alt = data.name;
+
+  cardElement.querySelector(".card__title").textContent = data.name;
+
+  setUpImagePreview(cardImageEl, data);
+  setUpLikeButton(cardElement.querySelector(".card__like-btn"), data);
+  cardElement
+    .querySelector(".card__delete-btn")
+    .addEventListener("click", () => openDeleteModal(cardElement, data._id));
+
+  return cardElement;
+}
+
+// Helper function to set up image preview
+function setUpImagePreview(imageElement, data) {
+  imageElement.addEventListener("click", () => {
+    modalImage.src = data.link;
+    modalImage.alt = data.name;
+    modalCaption.textContent = data.name;
+
+    openModal(previewModal);
+  });
+}
+
+// Function for Like button handler
+function setUpLikeButton(likeButton, data) {
+  if (data.isLiked) likeButton.classList.add("card__like-btn_liked");
+
+  likeButton.addEventListener("click", () => {
+    const isLiked = likeButton.classList.contains("card__like-btn_liked");
+    api
+      .handleLikeStatus(data._id, isLiked)
+      .then((updatedData) =>
+        likeButton.classList.toggle("card__like-btn_liked", updatedData.isLiked)
+      )
+      .catch((error) => console.error("Failed to update like status:", error));
+  });
+}
+
+// Function for Delete modal logic
+function openDeleteModal(cardElement, cardId) {
+  selectedCard = cardElement;
+  selectedCardId = cardId;
+  openModal(deleteModal); // Open the delete modal
+}
+
+// Define the delete form submit handler only once
+const handleDeleteSubmit = (evt) => {
+  evt.preventDefault();
+  const deleteButton = evt.submitter; // Moved inside the function to access `evt.submitter`
+  showLoading(deleteButton, true, "Deleting...", "Delete");
+
+  api
+    .deleteCard(selectedCardId)
+    .then(() => {
+      selectedCard.remove();
+      closeModal(deleteModal);
+    })
+    .catch((error) => console.error("Failed to delete card:", error))
+    .finally(() => showLoading(deleteButton, false, "Deleteing", "Delete..."));
```

Comment:
* The `confirmation` button text is not correct after submission https://snipboard.io/kVIEbd.jpg 


It should be like this

```js


    .finally(() => showLoading(deleteButton, false, "Deleting...", "Delete"));

```


I'll not reject. Please, correct it